src/main/kotlin/nomic/domain/auth/TokenRegistry.kt

```diff
@@ -0,0 +1,106 @@
+     * @return The raw JWT Token string authenticating that user
+     */
+    fun issueToken(user: User, claims: Map<String, String> = mapOf()): String
+
+    /**
+     * Validates the authenticity of a specified raw JWT Token
+     *
+     * @param[rawToken] The raw string representing a JWT Token to be verified as authentic and valid
+     * @return A result object containing the success state, the subject, and all valid claims
+     */
+    fun validateToken(rawToken: String): TokenValidationResult
+}
+
+/**
+ * A data object containing the success state of a token validation request and relevent details about the token
+ *
+ * @property[isSuccess] The success state of the token validation request: true only if successfully validated
+ * @property[subject] If successful, the subject of the JWT Token, which is the user domain entity of the authenticated user; otherwise null
+ * @property[validClaims] A mapping of claims between string name and a stringified value; if unsuccessful, empty
+ */
+data class TokenValidationResult(val isSuccess: Boolean, val subject: User? = null, val validClaims: Map<String, String> = mapOf())
+
+/**
+ * This implementation of ITokenRegistry offers a straightforward implementation using RSA for token signing.
+ *
+ * @see[ITokenRegistry]
+ * @param[keyProvider] This dependency is used to retrieve the RSA keys used to sign all JWT Tokens to validate authenticity and integrity
+ * @param[usersRepo] This dependency is used to retrieve the user entity that is the subject of valid JWT Tokens
+ */
+@Service
+class TokenRegistry(
+    private val keyProvider: IKeyProvider,
+    private val usersRepo: IUserRepository
+) : ITokenRegistry {
+    private val algorithm: Algorithm
+    private val verifier: JWTVerifier
+
+    init {
+        val keyPair = keyProvider.getKeyPair()
+        algorithm = Algorithm.RSA256(keyPair.publicKey, keyPair.privateKey)
+        verifier = JWT.require(algorithm)
+            .withIssuer(JWT_ISSUER)
+            .withAudience(JWT_ISSUER)
+            .withClaimPresence(RegisteredClaims.SUBJECT)
+            .acceptLeeway(10) // Issued at, Expires at, and not before as a 10 second window of leeway
+            .build()
+    }
+
+    override fun issueToken(user: User, claims: Map<String, String>): String {
+        val tokenBuilder = JWT.create()
+            .withSubject(user.id.toString())
+            .withIssuer(JWT_ISSUER)
+            .withAudience(JWT_ISSUER)
+            .withIssuedAt(Instant.now())
+            .withExpiresAt(Instant.now().plus(Duration.ofHours(2)))
+            .withNotBefore(Instant.now())
+
+        for (claim in claims) {
+            tokenBuilder.withClaim(claim.key, claim.value)
+        }
+
+        return tokenBuilder.sign(algorithm)
+    }
+
+    override fun validateToken(rawToken: String): TokenValidationResult {
+        // TODO Refactor out the exception pattern, possibly swap out jwt libraries
```

Comment:
(It is of course possible that it might not be gotten around to, but who knows.)  <-- this line right here is really the only argument I need for this.  If you really feel this information needs to be tracked then we can discuss on Tuesday